pkg/sanity/sanity.go

```diff
@@ -449,3 +460,68 @@ func PseudoUUID() string {
 func UniqueString(prefix string) string {
 	return prefix + uniqueSuffix
 }
+
+// Return codes for CheckPath
+const (
+	PathIsFile     = "file"
+	PathIsDir      = "directory"
+	PathIsNotFound = "not_found"
+	PathIsOther    = "other"
+)
+
+// defaultCheckPath runs os.Stat against the provided path and returns
+// a code indicating whether it's a file, directory, not found, or other.
+// If an error occurs, it returns an empty string along with the error.
+func defaultCheckPath(path string) (string, error) {
+	var pathAttribute string
+	fi, err := os.Stat(path)
+	if err != nil {
+		if os.IsNotExist(err) {
+			return PathIsNotFound, nil
+		} else {
+			return "", err
+		}
+	}
+	switch mode := fi.Mode(); {
+	case mode.IsRegular():
+		pathAttribute = PathIsFile
+	case mode.IsDir():
+		pathAttribute = PathIsDir
+	default:
+		pathAttribute = PathIsOther
+	}
+	return pathAttribute, nil
+}
+
+// CheckPath takes a path parameter and returns a code indicating whether
+// it's a file, directory, not found, or other. This can be done using a
+// custom command, custom function, or by the defaultCheckPath function.
+// If an error occurs, it returns an empty string along with the error.
+func CheckPath(path string, config *TestConfig) (string, error) {
+	if path == "" {
+		return "", errors.New("path argument must not be empty")
+	}
+	if config == nil {
+		return "", errors.New("config argument must not be nil")
+	}
+
+	if config.CheckPathCmd != "" {
+		// Check the provided path using the check path command.
+		ctx, cancel := context.WithTimeout(context.Background(), config.CheckPathCmdTimeout)
+		defer cancel()
+
+		cmd := exec.CommandContext(ctx, config.CheckPathCmd, path)
+		cmd.Stderr = os.Stderr
+		out, err := cmd.Output()
+		if err != nil {
+			return "", fmt.Errorf("check path command %s failed: %v", config.CheckPathCmd, err)
+		}
+		// Convert the command's stdout to an integer. This is expected to match
+		// the value for PathIsFile, PathIsDir, PathIsNotFound, or PathIsOther.
+		outstr := strings.TrimSpace(string(out))
```

Comment:
Here it would be good to check that the script has return a valid string. If not, raise an error.

Reply:
Done